quil-rs/src/parser/gate.rs

```diff
@@ -35,15 +36,16 @@ pub(crate) fn parse_gate<'a>(input: ParserInput<'a>) -> InternalParserResult<'a,
     ))(input)?;
     let parameters = parameters.unwrap_or_default();
     let (input, qubits) = many0(common::parse_qubit)(input)?;
-    Ok((
-        input,
-        Instruction::Gate(Gate {
-            name,
-            parameters,
-            qubits,
-            modifiers,
-        }),
-    ))
+    let gate = Gate::new(name.as_str(), parameters, qubits, modifiers).map_err(|e| {
+        nom::Err::Error(InternalError::from_kind(
+            input,
+            ParserErrorKind::InvalidInstruction {
+                instruction_type: "Gate".to_string(),
```

Comment:
Not saying it couldn't happen, but I don't see another place this would be needed. These will be isolated to the point of creation of an instruction type from the parser.